libraries/Zigbee/src/ZigbeeEP.cpp

```diff
@@ -0,0 +1,170 @@
+        delete[] zb_name;
+        return;
+    }
+    char* zb_model = new char[length + 2];
+    zb_model[0] = static_cast<char>(length);
+    memcpy(zb_model + 1, model, length);
+    zb_model[length + 1] = '\0';
+
+    // Get the basic cluster and update the manufacturer and model attributes
+    esp_zb_attribute_list_t *basic_cluster = esp_zb_cluster_list_get_cluster(_cluster_list, ESP_ZB_ZCL_CLUSTER_ID_BASIC, ESP_ZB_ZCL_CLUSTER_SERVER_ROLE);
+    esp_zb_basic_cluster_add_attr(basic_cluster, ESP_ZB_ZCL_ATTR_BASIC_MANUFACTURER_NAME_ID, (void *)zb_name);
+    esp_zb_basic_cluster_add_attr(basic_cluster, ESP_ZB_ZCL_ATTR_BASIC_MODEL_IDENTIFIER_ID, (void *)zb_model);
+}
+
+char* ZigbeeEP::readManufacturer(uint8_t endpoint, uint16_t short_addr) {
+    /* Read peer Manufacture Name & Model Identifier */
+    esp_zb_zcl_read_attr_cmd_t read_req;
+    read_req.address_mode = ESP_ZB_APS_ADDR_MODE_16_ENDP_PRESENT;
+    read_req.zcl_basic_cmd.src_endpoint = _endpoint;
+    read_req.zcl_basic_cmd.dst_endpoint = endpoint;
+    read_req.zcl_basic_cmd.dst_addr_u.addr_short = short_addr;
+    read_req.clusterID = ESP_ZB_ZCL_CLUSTER_ID_BASIC;
+
+    uint16_t attributes[] = {
+    ESP_ZB_ZCL_ATTR_BASIC_MANUFACTURER_NAME_ID,
+    };
+    read_req.attr_number = ZB_ARRAY_LENTH(attributes);
+    read_req.attr_field = attributes;
+
+
+    // clear read manufacturer
+    _read_manufacturer = nullptr;
+
+    esp_zb_zcl_read_attr_cmd_req(&read_req);
+
+    //Wait for response or timeout
+    if(xSemaphoreTake(lock, portMAX_DELAY) != pdTRUE){
+        log_e("Error while reading manufacturer");
+    }
+    return _read_manufacturer;
+
+}
+
+char* ZigbeeEP::readModel(uint8_t endpoint, uint16_t short_addr) {
+    /* Read peer Manufacture Name & Model Identifier */
+    esp_zb_zcl_read_attr_cmd_t read_req;
+    read_req.address_mode = ESP_ZB_APS_ADDR_MODE_16_ENDP_PRESENT;
+    read_req.zcl_basic_cmd.src_endpoint = _endpoint;
+    read_req.zcl_basic_cmd.dst_endpoint = endpoint;
+    read_req.zcl_basic_cmd.dst_addr_u.addr_short = short_addr;
+    read_req.clusterID = ESP_ZB_ZCL_CLUSTER_ID_BASIC;
+
+    uint16_t attributes[] = {
+    ESP_ZB_ZCL_ATTR_BASIC_MODEL_IDENTIFIER_ID,
+    };
+    read_req.attr_number = ZB_ARRAY_LENTH(attributes);
+    read_req.attr_field = attributes;
+
+    // clear read model
+    _read_model = nullptr;
+
+    esp_zb_zcl_read_attr_cmd_req(&read_req);
+
+    //Wait for response or timeout
+    //Semaphore take
+    if(xSemaphoreTake(lock, portMAX_DELAY) != pdTRUE){
+        log_e("Error while reading model");
+    }
+    return _read_model;
+}
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-variable"
+
+void ZigbeeEP::printBoundDevices() {
+    log_i("Bound devices:");
+    for(const auto& device : _bound_devices) {
+        log_i("Device on endpoint %d, short address: 0x%x", device->endpoint, device->short_addr);
+        print_ieee_addr(device->ieee_addr);
+    }
+}
+
+#pragma GCC diagnostic pop
```

Comment:
What variable gives the unused error ? Maybe we can use ´[[maybe_unused]]´ instead.

Reply:
If I remember correctly it was the `device` variable. I did retest but was not able to reproduce now.

Reply:
```
/home/runner/Arduino/hardware/espressif/esp32/libraries/Zigbee/src/ZigbeeEP.cpp:129:21: error: unused variable 'device' [-Werror=unused-variable]
  129 |     for(const auto& device : _bound_devices) {
      |                     ^~~~~~
cc1plus: some warnings being treated as errors
```

Reply:
changed to your suggestion :)